cmd/snap-update-ns/change.go

```diff
@@ -121,6 +120,25 @@ func (c *Change) createPath(path string, pokeHoles bool, as *Assumptions) ([]*Ch
 		err = MkfileAll(path, mode, uid, gid, rs)
 	case "symlink":
 		err = MksymlinkAll(path, mode, uid, gid, c.Entry.XSnapdSymlink(), rs)
+	case "ensure-dir":
+		uid = sysGetuid()
+		gid = sysGetgid()
+		if uid == 0 {
+			logger.Noticef("WARNING: cannot perform ensure-dir mount for target %q, not supported for the root user", path)
```

Comment:
I'm still slightly confused here because what snap-confine does is not support a user mount namespace for too, but this is disabling ensure-dir for any kind of namespace but if I remember you didn't make ensure-dir user mount namespace only?

Reply:
As discussed, this check is not necessary. Removed the check to enable ensure-dir for root/non-root, even though currently it's only used for user-mounts.